systemservices/marketplace/src/tasks/publishPurchase.ts

```diff
@@ -0,0 +1,24 @@
+import { TaskInputs, TaskOutputs } from "mesg-js/lib/service"
+import Web3 from "web3"
+import { decodeLog } from "../contracts/utils";
+import { eventHandlers } from "../events";
+
+export default (
+  web3: Web3,
+) => async (inputs: TaskInputs, outputs: TaskOutputs): Promise<void> => {
+  try {
+    let receipt = null
+    for (let i=0; i<inputs.signedTransactions.length; i++) {
+      receipt = await web3.eth.sendSignedTransaction(inputs.signedTransactions[i])
```

Comment:
why multiple tx here but we collect only the results from the last one

Reply:
There is either 1 or 2 transaction.
The optional first one is the erc20 approval of the marketplace to spend token on behalf of user.
The second, is the actual purchase of the service.
Only the last one is the important one.